HackingWithSwift/SwiftUI/myProject2/myProject2/myProject2/ContentView.swift

```diff
@@ -0,0 +1,201 @@
+                }
+            }
+            .alert("Game over", isPresented: $endGame) {
+                Button ("Restart") {}
+            } message: {
+                Text("\(finalScore)")
+            }
+            .navigationTitle("Rock, Paper, Scissons!")
+        }
+    }
+    
+    func compChoose() -> some View {
+        Text(optionToСhoice[computerСhoice])
+            .font(.largeTitle.weight(.semibold))
+    }
+    
+    func rock()  {
+        if computerСhoice == 0 {
+            myChoise = "✊"
+            score += 1
+            newGame()
+            rightСhoice = true
+        }
+        else if computerСhoice == 1 {
+            myChoise = "✊"
+            score -= 1
+            newGame()
+        } else {
+            myChoise = "✊"
+            score += 1
+            newGame()
+            rightСhoice = true
+        }
+    }
+    
+    func paper() {
+        if computerСhoice == 0 {
+            myChoise = "✋"
+            score += 1
+            newGame()
+            rightСhoice = true
+        }
+        else if computerСhoice == 1 {
+            myChoise = "✋"
+            score += 1
+            newGame()
+            rightСhoice = true
+        } else {
+            myChoise = "✋"
+            score -= 1
+            newGame()
+        }
+    }
+    
+    func scissons() {
+        if computerСhoice == 0 {
+            myChoise = "✌️"
+            score -= 1
+            newGame()
+        }
+        else if computerСhoice == 1 {
+            myChoise = "✌️"
+            score += 1
+            newGame()
+            rightСhoice = true
+        } else {
+            myChoise = "✌️"
+            score += 1
+            newGame()
+            rightСhoice = true
+        }
+    }
+    
+    func myturn() -> some View {
+        Text("\(myChoise)")
+            .font(.system(size: 150))
+            .padding(25)
+            .background(Color.secondary)
+            .clipShape(.circle)
+            .shadow(radius: 40)
+    }
+    
+    func goodChoise() -> some View {
+        if rightСhoice == true {
+            Text("You win")
+        } else {
+            Text("You lose")
+        }
+    }
+    
+    func newGame () {
+        computerСhoice = Int.random(in: 0...2)
+        rightСhoice = false
+        if counter == 10 {
+            restart()
+        }
+        
+        func restart() {
+            endGame = true
+            finalScore = "Your score is \(score)"
+            resetGame()
+        }
+        
+        func resetGame () {
+            score = 0
+            counter = 0
+        }
+    }
+}
+//        func button() -> some View {
+//            ForEach(0..<3) { number in
+//                Button {
+//                } label: {
+//                    Text(optionToСhoice[number])
+//        .padding(20)
+//        .background(.black)
+//        .foregroundStyle(.white)
+//        .clipShape(.buttonBorder)
+//        .font(.system(size: 70))
+//                }
+//            }
+//        }
+
```

Comment:
You're close to getting `ForEach` right! Here are a few tips:

1. You need to group the items horizontally.
2. Use `index` instead of `number`—it explains better what it is.
3. Add an action to each button. Hint: use the `index` to tell which button was pressed and call the right function.